--- tmp/iRODS/server/api/src/rsExecCmd.c	2013-05-17 21:56:10.000000000 +0200
+++ iRODS/server/api/src/rsExecCmd.c	2014-01-22 11:07:51.281182569 +0100
@@ -67,9 +67,13 @@
     ruleExecInfo_t rei;
 
     initReiWithDataObjInp (&rei, rsComm, NULL);
-    char *args[1];
+    char *args[4];
     args[0] = execCmdInp->cmd;
-    status = applyRuleArg ("acPreProcForExecCmd", args, 1, &rei, NO_SAVE_REI);
+    args[1] = execCmdInp->cmdArgv == NULL? (char*)"" : execCmdInp->cmdArgv;
+    args[2] = execCmdInp->execAddr == NULL ? (char*)"" : execCmdInp->execAddr;
+    args[3] = execCmdInp->hintPath == NULL ? (char*)"" : execCmdInp->hintPath;
+    status = applyRuleArg ("acPreProcForExecCmd", args, 4, &rei, NO_SAVE_REI);
+
     if (status < 0) {
         rodsLog (LOG_ERROR,
                  "initAgent: acPreProcForExecCmd error, status = %d", status);
